frontend/frontend.go

```diff
@@ -46,9 +48,11 @@ func MuxPattern(method string, segments ...string) string {
 }
 
 func NewFrontend(logger *slog.Logger, listener net.Listener) *Frontend {
+	prometheusEmitter := NewPrometheusEmitter()
```

Comment:
Maybe I am missing some important reason to do so, but similar to the previous comment, I think it would be great to pass an Emitter interface as a parameter to the _NewFrontend_ function (dependency injection) instead of hardcoding the value in the constructor. If we do so, we would go to frontend/main.go and there provide the actual implementation of the interface and inject it to the NewFrontEnd function.
// frontend/frontend.go
```go
func NewFrontend(logger *slog.Logger, listener net.Listener, emitter metrics.Emitter) *Frontend {
```
// frontend/main.go
```go
prometheusEmitter := NewPrometheusEmitter()
frontend := NewFrontend(logger, listener, prometheusEmitter)
```